src/App.js

```diff
@@ -1,19 +1,28 @@
 import React from "react";
-import { Routes, Route } from "react-router-dom";
-import { Navbar } from "./components";
+import { Routes, Route, Navigate } from "react-router-dom";
+import { Navbar, Toast } from "./components";
+import { useAuthContext } from "./context/AuthProvider";
+import { useStateContext } from "./context/StateProvider";
 import { useGetVideos } from "./hooks";
 import { useGetCategories } from "./hooks/useGetCategories";
-import { Explore, Home } from "./pages";
+import { Explore, Home, Login, Profile, SignUp, Error404 } from "./pages";
 
 export const App = () => {
   useGetCategories();
   useGetVideos();
+  const { toast } = useStateContext();
+  const { userState } = useAuthContext();
   return (
     <>
+      {toast.showToast && <Toast />}
```

Comment:
Maybe you can shift this showToast conditional rendering to toast component. just suggesting so it will look better